components/kyma-adapter/internal/handler/handler.go

```diff
@@ -1,16 +1,300 @@
+	"github.com/kyma-incubator/compass/components/director/pkg/log"
+	"github.com/pkg/errors"
+)
+
+const (
+	assignOperation   = "assign"
+	unassignOperation = "unassign"
+
+	readyState         string = "READY"
+	configPendingState string = "CONFIG_PENDING"
+	createErrorState   string = "CREATE_ERROR"
+	deleteErrorState   string = "DELETE_ERROR"
+
+	successStatusCode int = http.StatusOK
 )
 
-// AdapterHandler processes received requests
+// Client is responsible for making internal graphql calls to the director
+//go:generate mockery --name=Client --output=automock --outpkg=automock --case=underscore --disable-version-string
+type Client interface {
+	GetApplicationBundles(ctx context.Context, appID, tenant string) ([]*graphql.BundleExt, error)
+	CreateBasicBundleInstanceAuth(ctx context.Context, tenant, bndlID, rtmID, username, password string) error
+	CreateOauthBundleInstanceAuth(ctx context.Context, tenant, bndlID, rtmID, tokenServiceURL, clientID, clientSecret string) error
+	UpdateBasicBundleInstanceAuth(ctx context.Context, tenant, authID, bndlID, username, password string) error
+	UpdateOauthBundleInstanceAuth(ctx context.Context, tenant, authID, bndlID, tokenServiceURL, clientID, clientSecret string) error
+	DeleteBundleInstanceAuth(ctx context.Context, tenant, authID string) error
+}
+
+// AdapterHandler is the Kyma Tenant Mapping Adapter handler which processes the received requests
 type AdapterHandler struct {
-	DirectorGqlClient *graphql.Client
+	DirectorGqlClient Client
+}
+
+// NewHandler creates an AdapterHandler
+func NewHandler(directorGqlClient Client) *AdapterHandler {
+	return &AdapterHandler{DirectorGqlClient: directorGqlClient}
 }
 
 // HandlerFunc is the implementation of AdapterHandler
 func (a AdapterHandler) HandlerFunc(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	log.C(ctx).Info("The Kyma Tenant Mapping Adapter was hit.")
+
+	var reqBody types.Body
+	if err := decodeJSONBody(w, r, &reqBody); err != nil {
+		var mr *malformedRequest
+		if errors.As(err, &mr) {
+			respondWithError(ctx, w, mr.status, "", mr)
+		} else {
+			respondWithError(ctx, w, http.StatusInternalServerError, "", errors.Wrap(err, "while decoding json request body"))
+		}
+		return
+	}
+
+	log.C(ctx).Info("Validating tenant mapping request body...")
+	if err := reqBody.Validate(); err != nil {
+		respondWithError(ctx, w, http.StatusBadRequest, "", errors.Wrapf(err, "while validating the request body"))
+		return
+	}
+	tenantID := reqBody.ReceiverTenant.OwnerTenant
+	log.C(ctx).Infof("The request has tenant with id %q", tenantID)
+
+	if reqBody.Context.Operation == assignOperation {
+		log.C(ctx).Infof("The request operation is %q", assignOperation)
+		a.processAssignOperation(ctx, w, reqBody, tenantID)
+	} else {
+		log.C(ctx).Infof("The request operation is %q", unassignOperation)
+		a.processUnassignOperation(ctx, w, reqBody, tenantID)
+	}
+
+	log.C(ctx).Info("The Kyma integration was successfully processed")
```

Comment:
I am removing this log because on the `if..else..` above there are logs in the `respondWithError` or `respondWithSuccess` funcs and I think this log here is obsolete.